### PR TITLE
Proper support for users that are coming from Google

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
- prod configuration for Google
- as legacy Google does not have 'login', so we use email as a name and trim @gmail from it

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt [hunk 1]
```diff
@@ -221,13 +221,16 @@ class RegistrationView : AbstractView<RegistrationProps, RegistrationViewState>(
         "PARAMETER_NAME_IN_OUTER_LAMBDA",
     )
     private fun ChildrenBuilder.renderInputForm() {
+        // google does not provide us login, only e-mail, so need to trim everything after "@gmail"
+        val rawInput = state.fieldsMap[InputTypes.USER_NAME] ?: ""
+        val atIndex = rawInput.indexOf('@')
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:226]: do we forbid manual creating the name with `@`?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt [hunk 2]
```diff
@@ -221,13 +221,16 @@ class RegistrationView : AbstractView<RegistrationProps, RegistrationViewState>(
         "PARAMETER_NAME_IN_OUTER_LAMBDA",
     )
     private fun ChildrenBuilder.renderInputForm() {
+        // google does not provide us login, only e-mail, so need to trim everything after "@gmail"
+        val rawInput = state.fieldsMap[InputTypes.USER_NAME] ?: ""
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:225]: "" means guest? do we have some data class with such field (`GUEST`)

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt (hunk 1 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:226]: yes, moreover - you even cannot input "@" into the input field

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:225]: why not `.orEmpty()`?

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt (hunk 2 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:225]: very good suggestion, please create an issue for diktat @sanyavertolet 
We have a thousand of places where it is in save, will change everywhere

Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:225]: > very good suggestion, please create an issue for diktat @sanyavertolet
> We have a thousand of places where it is in save, will change everywhere

@sanyavertolet I will create

Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:225]: https://github.com/saveourtool/diktat/issues/1710

**Pull request merged**